### ISSUE TITLE
Add Creator permission to Users

### ISSUE THREAD

**Issue by fritosxii** [matrix-msu/kora#632]
Along with active and admin, add a permission to Users called Creator. This permission will allow a user to create projects, without giving them access to all projects, or giving them system administrator privileges

**Comment by watrall** [comment]
did this get put on the backburner?

**Comment by fritosxii** [comment]
Yes, because we still need to define roles and what those roles can do

**Comment by fritosxii** [comment]
3 models that pretty much handle everything (ProjectGroup, FormGroup, and permission funcitons in User)

Main process would be creating a new class (Model, DB and Controller), and overwriting above class references with respective new class functions. 

We then need to also modify Permission views for Projects and Forms, into a single view for new Project level permissions. User also has some views for viewing single user permissions that will need to be modified.

We will need some sort of conversion for old group types, to new group types

The big thing that needs to be done, is a feature-by-feature breakdown so we can deal with edge cases. This is step one for me.

Best case scenario 2-3 weeks once roles are defined
